Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM golang:1.16.9
+
+RUN apt-get update && \
+    apt-get install -y apt-utils \
+    zip \
+    unzip \
+    vim \
+    curl \
+    netcat \
+    git
+
+WORKDIR  /go/src/github.com/delta/dalal-street-server 
+
+ADD ./docker_server_setup.sh ./docker_server_setup.sh
+RUN "./docker_server_setup.sh"
+# The above setup is run seperately earlier on, so that cache can be used 
+# when rebuilding the image, in case of any change
+
```

Comment:
`COPY` go.mod go.sum ./

`RUN` go mod download

use this to install dependency

check [here](https://docs.docker.com/language/golang/build-images/) for more info